File: WowsKarma.Common/Models/DTOs/PlayerClanProfileDTO.cs
using Nodsoft.Wargaming.Api.Common.Data.Responses.Wows;

namespace WowsKarma.Common.Models.DTOs;

public record PlayerClanProfileDTO
{
	public ClanListingDTO ClanInfo { get; init; }
	
	public ClanRole ClanMemberRole { get; init; }
	
	// HACK: DateOnly / TimeOnly serialization is not supported by STJ as of now.
	// See: https://github.com/dotnet/runtime/issues/53539
	public DateTime JoinedClanAt { get; init; }
}

File: wowskarma.app/src/app/services/api/models/player-clan-profile-dto.ts
/* tslint:disable */
/* eslint-disable */
import { ClanListingDto } from './clan-listing-dto';
import { ClanRole } from './clan-role';
export interface PlayerClanProfileDto {
  clanInfo?: ClanListingDto;
  clanMemberRole?: ClanRole;
  joinedClanAt?: string;
}
